aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -744,15 +738,21 @@ TensorIterator TensorIterator::reduce_op(Tensor& out1, Tensor& out2, const Tenso
       " and output2 has ", out2.sizes());
   TORCH_CHECK(out1.strides() == out2.strides(), "reduce_op(): expected both outputs to have same strides, but output1 has ", out1.strides(),
       " and output2 has ", out2.strides());
-  auto iter = TensorIterator();
+  auto iter = TensorIteratorConfig();
   iter.add_output(out1);
   iter.add_output(out2);
   iter.add_input(a);
-  iter.config_resize_outputs_ = false;
-  iter.config_is_reduction_ = true;
+  iter.dont_resize_outputs();
+  iter.is_reduction(true);
   iter.check_all_same_dtype(false);
-  iter.build();
-  return iter;
+  return TensorIterator(iter);
+}
+
+void TensorIterator::populate_operands() {
```

Comment:
If you want to do something like this then you should require that the config be moved in the TI constructor, right? 
However, it'd be nice if you didn't have to copy into the TI at all. Would a pattern like config().build_iterator() -> TensorIterator be more natural?

Reply:
It's actually OK, moving doesn't help at all because TensorIteratorConfig is effectively a value type (because SmallVector inlines its storage into the struct, moving doesn't save you any memory traffic, you still have to move each of the individual entries; everything is just primitive types).

> However, it'd be nice if you didn't have to copy into the TI at all. Would a pattern like config().build_iterator() -> TensorIterator be more natural?

Maybe. It depends whether or not the configuration options are used outside of build. I will investigate this; if we can avoid depending on the configuration options beyond build that actually is a pretty big improvement.

Reply:
Sorry, I was being unclear. My suggestion to std::move the config into the TI if you follow this pattern was for safety.

Reply:
Constructing/destructing `operands_` is a noticeable part of TensorIterator ops, so we should be careful not to do it twice, in config and in TI itself.

Reply:
> My suggestion to std::move the config into the TI if you follow this pattern was for safety.

Yeah, I guess what I'm saying is that this doesn't actually help (because the move constructor on SmallVector is nearly just as expensive as the copy constructor).

> Constructing/destructing operands_ is a noticeable part of TensorIterator ops, so we should be careful not to do it twice, in config and in TI itself.

Duly noted